editoast/src/infra_cache/operation/delete.rs

```diff
@@ -90,217 +82,46 @@ mod tests {
-            .apply(infra.id, db_pool.get_ok().deref_mut())
-            .await
-            .is_ok());
-
-        let res_del = sql_query(format!(
-                "SELECT COUNT (*) AS nb FROM infra_object_buffer_stop WHERE obj_id = '{}' AND infra_id = {}",
-                buffer_stop.get_id(),
-                infra.id
-            ))
-            .get_result::<Count>(db_pool.get_ok().deref_mut()).await.unwrap();
-
-        assert_eq!(res_del.nb, 0);
-    }
-
-    #[rstest]
-    async fn delete_route() {
-        let db_pool = DbConnectionPoolV2::for_tests();
-        let infra = create_empty_infra(db_pool.get_ok().deref_mut()).await;
-        let route = create_route(db_pool.get_ok().deref_mut(), infra.id, Default::default()).await;
-        let route_deletion: DeleteOperation = route.get_ref().into();
-
-        assert!(route_deletion
-            .apply(infra.id, db_pool.get_ok().deref_mut())
-            .await
-            .is_ok());
-
-        let res_del = sql_query(format!(
-            "SELECT COUNT (*) AS nb FROM infra_object_route WHERE obj_id = '{}' AND infra_id = {}",
-            route.get_id(),
-            infra.id
-        ))
-        .get_result::<Count>(db_pool.get_ok().deref_mut())
-        .await
-        .unwrap();
-
-        assert_eq!(res_del.nb, 0);
-    }
-
-    #[rstest]
-    async fn delete_op() {
-        let db_pool = DbConnectionPoolV2::for_tests();
-        let infra = create_empty_infra(db_pool.get_ok().deref_mut()).await;
-        let op = create_op(db_pool.get_ok().deref_mut(), infra.id, Default::default()).await;
-
-        let op_deletion: DeleteOperation = op.get_ref().into();
-
-        assert!(op_deletion
-            .apply(infra.id, db_pool.get_ok().deref_mut())
-            .await
-            .is_ok());
-        let res_del = sql_query(format!(
-                "SELECT COUNT (*) AS nb FROM infra_object_operational_point WHERE obj_id = '{}' AND infra_id = {}",
-                op.get_id(),
-                infra.id
-            ))
-            .get_result::<Count>(db_pool.get_ok().deref_mut()).await.unwrap();
-
-        assert_eq!(res_del.nb, 0);
+    macro_rules! test_delete_object {
+        ($obj:ident) => {
+            paste::paste! {
+                #[rstest::rstest]
+                async fn [<test_delete_$obj:snake>]() {
+                    use diesel_async::RunQueryDsl;
+                    use std::ops::DerefMut;
+
+                    let db_pool = crate::modelsv2::DbConnectionPoolV2::for_tests();
+                    let infra = crate::modelsv2::fixtures::create_empty_infra(db_pool.get_ok().deref_mut()).await;
+
+                    let railjson_object = crate::infra_cache::operation::create::RailjsonObject::$obj {
+                        railjson: $obj::default(),
+                    };
+                    assert!(crate::infra_cache::operation::create::apply_create_operation(&railjson_object, infra.id, db_pool.get_ok().deref_mut()).await.is_ok());
```

Comment:
Since this is generated code it's important to add a message for why an assertion failed as it can be quite tricky to identify where the faulty `assert` is. Especially when there are more than one.

```suggestion
                    let result = crate::infra_cache::operation::create::apply_create_operation(&railjson_object, infra.id, db_pool.get_ok().deref_mut()).await;
                    assert!(result.is_ok(), "failed to create a {}", stringify!($obj));
```